conftest.py

```diff
@@ -11,103 +9,34 @@
 os.environ["TF_CPP_MIN_LOG_LEVEL"] = "2"
 logging.basicConfig(level=logging.ERROR)
 
-from hub.api.dataset import Dataset
-from hub.constants import (
-    MIN_FIRST_CACHE_SIZE,
-    MIN_SECOND_CACHE_SIZE,
-    PYTEST_LOCAL_PROVIDER_BASE_ROOT,
-    PYTEST_MEMORY_PROVIDER_BASE_ROOT,
-    PYTEST_S3_PROVIDER_BASE_ROOT,
-)
-from hub.core.storage import LocalProvider, MemoryProvider, S3Provider
-from hub.core.tests.common import LOCAL, MEMORY, S3
-from hub.tests.common import SESSION_ID, current_test_name, get_dummy_data_path
-from hub.util.cache_chain import get_cache_chain
+from hub.constants import *
+from hub.tests.common import SESSION_ID
 
-
-MEMORY_OPT = "--memory-skip"
-LOCAL_OPT = "--local"
-S3_OPT = "--s3"
-CACHE_OPT = "--cache-chains"
-CACHE_ONLY_OPT = "--cache-chains-only"
-S3_PATH_OPT = "--s3-path"
-KEEP_STORAGE_OPT = "--keep-storage"
-FULL_BENCHMARK_OPT = "--full-benchmarks"
-
-# @pytest.mark.`FULL_BENCHMARK_MARK` is how full benchmarks are notated
-FULL_BENCHMARK_MARK = "full_benchmark"
-
-
-def _get_storage_configs(request):
-    return {
-        MEMORY: {
-            "base_root": PYTEST_MEMORY_PROVIDER_BASE_ROOT,
-            "class": MemoryProvider,
-            "use_id": False,
-            "is_id_prefix": False,
-            # if is_id_prefix (and use_id=True), the session id comes before test name, otherwise it is reversed
-        },
-        LOCAL: {
-            "base_root": PYTEST_LOCAL_PROVIDER_BASE_ROOT,
-            "class": LocalProvider,
-            "use_id": False,
-            "is_id_prefix": False,
-        },
-        S3: {
-            "base_root": request.config.getoption(S3_PATH_OPT),
-            "class": S3Provider,
-            "use_id": True,
-            "is_id_prefix": True,
-        },
-    }
-
-
-def _is_opt_true(request, opt):
-    return request.config.getoption(opt)
+# import * so all fixtures can be used accross the project
+from hub.tests.path_fixtures import *
+from hub.tests.dataset_fixtures import *
+from hub.tests.storage_fixtures import *
+from hub.tests.cache_fixtures import *
+from hub.tests.client_fixtures import *
 
 
 def pytest_addoption(parser):
     parser.addoption(
-        MEMORY_OPT,
-        action="store_true",
-        help="Tests using the `memory_storage` fixture will be skipped. Tests using the `storage` fixture will be "
-        "skipped if called with `MemoryProvider`.",
+        MEMORY_OPT, action="store_true", help="Memory tests will be SKIPPED if enabled."
```

Comment:
store true won't really mean anything for memory datasets I guess. Maybe we can mention that somewhere?

Reply:
not sure what you mean, `--memory-skip` is the `MEMORY_OPT` and when you call the tests with `--memory-skip`, all memory tests are skipped.

Reply:
I meant --keep-storage doesn't work for memory datasets, can we mention that somewhere?

Reply:
yep did this in CONTRIBUTING.md and in the conftest